### PR TITLE
pkg/trace/sampler: Makes Rare Sampler more configurable

### PR DESCRIPTION
**Pull request by Barbayar**
### What does this PR do?
- makes RareSampler more configurable
- updates some default configuration values

|         **Configuration**         | **Old Default** | **New Default** | **Config Name**                       |
|:---------------------------------|---------------:|---------------:|---------------------------------------:|
| Cool down for non priority traces |           2 min |           5 min | `apm_config.rare_sampler.cooldown`    |
| TPS                               |               5 |               5 | `apm_config.rare_sampler.tps`         |
| Cardinality                       |            1000 |             200 | `apm_config.rare_sampler.cardinality` |

### Motivation
Rare Sampler samples too much data traces for data storage services. Data storage services have high cardinality resources. As result, Rare Sampler ends up annotating too many resources as rare.
With this change, the rare sampler should sample less traces when the resource cardinality is high.

### Additional Notes
N/A

### Possible Drawbacks / Trade-offs
If `env/service` has more than 200 resources, there will be less chances to catch actual rare traces. But, it knobs can be configured, now.

### Describe how to test/QA your changes
We should notice less or the same amount of rare traces ingested comparing to the previous version.

### Reviewer's Checklist

- [x] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [x] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [x] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [x] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [x] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [x] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [x] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Comment by ajgajg1134** [comment]
Can you add some release notes for this change? Also does there need to be an associated documentation change somewhere else?

**Review by dumontg — APPROVED** [review]
LGTM

**Review by buraizu — APPROVED** [review]

**Review by knusbaum — COMMENTED** [review]
@Barbayar @ajgajg1134 @gbbr 

Do we want to add these configs to the config template? https://github.com/DataDog/datadog-agent/blob/7b7b9d506ebe47f27527bce7e240f72f68959293/pkg/config/config_template.yaml#L1003-L1179

I dislike when there are config options that are documented nowhere except the code.

**Comment by gbbr** [comment]
Probably not but I'll let Barbayar confirm. I think that most cases users should not touch any of this.

**Comment by Barbayar** [comment]
@gbbr Yes. The users shouldn't touch these configs, unless the support team suggests them to update.

The defaults values are intended to work for all users.

File: pkg/trace/config/config.go [hunk 1]
```diff
@@ -289,12 +289,17 @@ type AgentConfig struct {
...
-	MaxEPS             float64
-	MaxRemoteTPS       float64
+	ExtraSampleRate float64
+	TargetTPS       float64
+	ErrorTPS        float64
+	MaxEPS          float64
+	MaxRemoteTPS    float64
+
+	// Rare Sampler configuation
+	DisableRareSampler        bool
```
Comment (gbbr) [inline pkg/trace/config/config.go:299]:
```suggestion
	RareSamplerDisabled        bool
```
Just to be on par with all the others :)

**Review by gbbr — APPROVED** [review]

**Pull request merged**